@myarjunar I tested it with EU (using US credentials) and it does not gracefully handle invalid credentials:

<img width="1413" alt="image" src="https://user-images.githubusercontent.com/178003/54957985-3661d200-4f4c-11e9-900c-c7a3a614bf7d.png">

```
An error has occurred while executing Python code: 

Exception: Authentication failed. 
Traceback (most recent call last):
  File "/Users/timlinux/Library/Application Support/QGIS/QGIS3/profiles/Maceira/python/plugins/geosys-qgis-plugin/geosys/ui/widgets/options_dialog.py", line 75, in request_token
    use_testing_service=self.use_testing_service())
  File "/Users/timlinux/Library/Application Support/QGIS/QGIS3/profiles/Maceira/python/plugins/geosys-qgis-plugin/geosys/bridge_api_wrapper.py", line 86, in __init__
    raise Exception(self.authentication_message)
Exception: Authentication failed.
```

Can you please make sure it handles invalid credentials well.

Also @AndreLester can you ask GeoSYS for test credentials for EU so we can test both use cases. Please add them to the secrets file.

--- a/geosys/bridge_api_wrapper.py
+++ b/geosys/bridge_api_wrapper.py
@@ -82,8 +82,6 @@ class BridgeAPI(object):
 
         # authenticate user
         self.authenticated, self.authentication_message = self.authenticate()
-        if not self.authenticated:
-            raise Exception(self.authentication_message)
 
     @staticmethod
     def get_crops():
@@ -124,7 +122,9 @@ class BridgeAPI(object):
                 message = 'Authentication succeeded.'
                 return True, message
             else:
-                message = 'Authentication failed.'
+                message = (
+                    'Authentication failed. Ensure your username and password '
+                    'are valid for the selected region service.')
                 return False, message
         except KeyError:
             message = 'Please enter a correct region (NA or EU)'

--- a/geosys/ui/templates/geosys_dockwidget_base.ui
+++ b/geosys/ui/templates/geosys_dockwidget_base.ui
@@ -103,7 +103,7 @@
     <item row="1" column="0">
      <widget class="QStackedWidget" name="stacked_widget">
       <property name="currentIndex">
-       <number>0</number>
+       <number>1</number>
       </property>
       <widget class="QWidget" name="coverage_form_page">
        <property name="minimumSize">
@@ -389,6 +389,21 @@
    <header>qgis.gui</header>
   </customwidget>
  </customwidgets>
+ <tabstops>
+  <tabstop>geometry_combo_box</tabstop>
+  <tabstop>map_product_combo_box</tabstop>
+  <tabstop>sensor_combo_box</tabstop>
+  <tabstop>start_date_edit</tabstop>
+  <tabstop>end_date_edit</tabstop>
+  <tabstop>cancel_push_button</tabstop>
+  <tabstop>help_push_button</tabstop>
+  <tabstop>back_push_button</tabstop>
+  <tabstop>next_push_button</tabstop>
+  <tabstop>coverage_result_search</tabstop>
+  <tabstop>coverage_result_list</tabstop>
+  <tabstop>raster_radio_button</tabstop>
+  <tabstop>vecto_radio_button</tabstop>
+ </tabstops>
  <resources/>
  <connections/>
 </ui>

--- a/geosys/ui/templates/geosys_dockwidget_base.ui
+++ b/geosys/ui/templates/geosys_dockwidget_base.ui
@@ -103,7 +103,7 @@
     <item row="1" column="0">
      <widget class="QStackedWidget" name="stacked_widget">
       <property name="currentIndex">
-       <number>1</number>
+       <number>0</number>
       </property>
       <widget class="QWidget" name="coverage_form_page">
        <property name="minimumSize">

--- a/geosys/ui/templates/options_dialog_base.ui
+++ b/geosys/ui/templates/options_dialog_base.ui
@@ -309,6 +309,16 @@
    </widget>
   </widget>
  </widget>
+ <tabstops>
+  <tabstop>username_form</tabstop>
+  <tabstop>password_form</tabstop>
+  <tabstop>save_credentials_checkbox</tabstop>
+  <tabstop>connect_button</tabstop>
+  <tabstop>us_radio_button</tabstop>
+  <tabstop>eu_radio_button</tabstop>
+  <tabstop>testing_service_checkbox</tabstop>
+  <tabstop>page_limit_form</tabstop>
+ </tabstops>
  <resources/>
  <connections>
   <connection>

--- a/geosys/ui/templates/options_dialog_base.ui
+++ b/geosys/ui/templates/options_dialog_base.ui
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>566</width>
-    <height>568</height>
+    <height>710</height>
    </rect>
   </property>
   <property name="minimumSize">
@@ -23,7 +23,7 @@
    <property name="geometry">
     <rect>
      <x>30</x>
-     <y>520</y>
+     <y>670</y>
      <width>511</width>
      <height>32</height>
     </rect>
@@ -39,7 +39,7 @@
    <property name="geometry">
     <rect>
      <x>270</x>
-     <y>490</y>
+     <y>640</y>
      <width>271</width>
      <height>20</height>
     </rect>
@@ -76,7 +76,7 @@
      </rect>
     </property>
     <property name="text">
-     <string>Set the paging limit for the maximum responses to be returned for each request. Larger limits will result in longer processing times and possible API timeouts.</string>
+     <string>Use valid credentials from GEOSYS. The credentials are required to get the access token which will be used for every Bridge API request.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
@@ -254,7 +254,7 @@
    <property name="geometry">
     <rect>
      <x>21</x>
-     <y>360</y>
+     <y>510</y>
      <width>519</width>
      <height>121</height>
     </rect>
@@ -308,6 +308,90 @@
     </layout>
    </widget>
   </widget>
+  <widget class="QGroupBox" name="coverage_parameters_group">
+   <property name="geometry">
+    <rect>
+     <x>20</x>
+     <y>360</y>
+     <width>521</width>
+     <height>151</height>
+    </rect>
+   </property>
+   <property name="title">
+    <string>Coverage Parameters</string>
+   </property>
+   <widget class="QLabel" name="sowing_date_label">
+    <property name="geometry">
+     <rect>
+      <x>40</x>
+      <y>111</y>
+      <width>91</width>
+      <height>21</height>
+     </rect>
+    </property>
+    <property name="text">
+     <string>Sowing Date</string>
+    </property>
+   </widget>
+   <widget class="QLabel" name="crop_label">
+    <property name="geometry">
+     <rect>
+      <x>40</x>
+      <y>80</y>
+      <width>71</width>
+      <height>21</height>
+     </rect>
+    </property>
+    <property name="text">
+     <string>Crop Type</string>
+    </property>
+   </widget>
+   <widget class="QComboBox" name="crop_combo_box">
+    <property name="geometry">
+     <rect>
+      <x>140</x>
+      <y>81</y>
+      <width>341</width>
+      <height>21</height>
+     </rect>
+    </property>
+    <property name="insertPolicy">
+     <enum>QComboBox::InsertAlphabetically</enum>
+    </property>
+   </widget>
+   <widget class="QDateEdit" name="sowing_date_edit">
+    <property name="geometry">
+     <rect>
+      <x>140</x>
+      <y>111</y>
+      <width>341</width>
+      <height>21</height>
+     </rect>
+    </property>
+    <property name="displayFormat">
+     <string>yyyy-MM-dd</string>
+    </property>
+    <property name="calendarPopup">
+     <bool>true</bool>
+    </property>
+   </widget>
+   <widget class="QLabel" name="coverage_parameters_label">
+    <property name="geometry">
+     <rect>
+      <x>30</x>
+      <y>40</y>
+      <width>471</width>
+      <height>31</height>
+     </rect>
+    </property>
+    <property name="text">
+     <string>Set the desired coverage parameters for each request. This parameters are required and will be applied to each coverage request</string>
+    </property>
+    <property name="wordWrap">
+     <bool>true</bool>
+    </property>
+   </widget>
+  </widget>
  </widget>
  <tabstops>
   <tabstop>username_form</tabstop>
@@ -317,6 +401,8 @@
   <tabstop>us_radio_button</tabstop>
   <tabstop>eu_radio_button</tabstop>
   <tabstop>testing_service_checkbox</tabstop>
+  <tabstop>crop_combo_box</tabstop>
+  <tabstop>sowing_date_edit</tabstop>
   <tabstop>page_limit_form</tabstop>
  </tabstops>
  <resources/>

--- a/geosys/ui/widgets/options_dialog.py
+++ b/geosys/ui/widgets/options_dialog.py
@@ -2,6 +2,7 @@
 """Implementation of GEOSYS options dialog.
 """
 from PyQt5 import QtGui, QtWidgets
+from PyQt5.QtWidgets import QMessageBox
 from qgis.PyQt.QtCore import QSettings
 
 from geosys.bridge_api_wrapper import BridgeAPI
@@ -68,12 +69,19 @@ class GeosysOptionsDialog(QtWidgets.QDialog, FORM_CLASS):
 
     def request_token(self):
         """Request access token to GEOSYS's identity server."""
-        bridge_api = BridgeAPI(
-            username=self.username(),
-            password=self.password(),
-            region=self.region(),
-            use_testing_service=self.use_testing_service())
+        message_title = "Bridge API Authentication Status"
+        try:
+            bridge_api = BridgeAPI(
+                username=self.username(),
+                password=self.password(),
+                region=self.region(),
+                use_testing_service=self.use_testing_service())
 
-        QtWidgets.QMessageBox.information(
-            self, "Bridge API Authentication Status",
-            bridge_api.authentication_message)
+            if not bridge_api.authenticated:
+                QMessageBox.critical(
+                    self, message_title, bridge_api.authentication_message)
+            else:
+                QMessageBox.information(
+                    self, message_title, bridge_api.authentication_message)
+        except Exception as e:
+            QMessageBox.critical(self, message_title, str(e))

